src/main/java/com/synopsys/integration/detect/lifecycle/run/RunManager.java

```diff
@@ -355,11 +387,11 @@ private void runBlackDuckProduct(ProductRunData productRunData, DetectConfigurat
         logger.debug("Completed Detect Code Location processing.");
 
         logger.info(ReportConstants.RUN_SEPARATOR);
-        if (detectToolFilter.shouldInclude(DetectTool.SIGNATURE_SCAN)) {
+        if ((detectToolFilter.shouldInclude(DetectTool.SIGNATURE_SCAN) || detectToolFilter.shouldInclude(DetectTool.DOCKER)) && enabledTools.contains(DetectTool.SIGNATURE_SCAN)) {
```

Comment:
I'm not sure what you mean by "double check" since I don't believe shouldInclude takes whether a tool is enabled into account at all (or is this building off your above comment, and you're suggesting we pass the list of enabled tools to a filter and on the include check that list is also referenced?).  I like the idea of adding a log to explain the "you tried to include this tool but you don't have permission"

Reply:
I think there should be a single shouldInclude(SIGNATURE_SCAN), no mention of DOCKER. And the feature enablement check on the next line.